Marlin/configuration_store.cpp

```diff
@@ -702,32 +716,33 @@ void MarlinSettings::postprocess() {
         EEPROM_READ(dummy);
       #endif
 
+      uint8_t grid_max_x, grid_max_y;
+      EEPROM_READ(grid_max_x);
+      EEPROM_READ(grid_max_y);
+
       //
       // Mesh (Manual) Bed Leveling
       //
 
       bool leveling_is_on;
-      uint8_t mesh_num_x, mesh_num_y;
       EEPROM_READ(leveling_is_on);
       EEPROM_READ(dummy);
-      EEPROM_READ(mesh_num_x);
-      EEPROM_READ(mesh_num_y);
 
       #if ENABLED(MESH_BED_LEVELING)
         mbl.status = leveling_is_on ? _BV(MBL_STATUS_HAS_MESH_BIT) : 0;
         mbl.z_offset = dummy;
-        if (mesh_num_x == GRID_MAX_POINTS_X && mesh_num_y == GRID_MAX_POINTS_Y) {
+        if (grid_max_x == GRID_MAX_POINTS_X && grid_max_y == GRID_MAX_POINTS_Y) {
           // EEPROM data fits the current mesh
           EEPROM_READ(mbl.z_values);
         }
         else {
           // EEPROM data is stale
           mbl.reset();
-          for (uint16_t q = mesh_num_x * mesh_num_y; q--;) EEPROM_READ(dummy);
+          for (uint16_t q = grid_max_x * grid_max_y; q--;) EEPROM_READ(dummy);
         }
       #else
-        // MBL is disabled - skip the stored data
-        for (uint16_t q = mesh_num_x * mesh_num_y; q--;) EEPROM_READ(dummy);
+        // MBL is disabled - skip the stored data by reading a default mesh
+        for (uint16_t q = defmeshdim * defmeshdim; q--;) EEPROM_READ(dummy);
```

Comment:
Yes, same logic as above -- this was changed because I changed to a global max value.  

Also, I created defmeshdim because there were a bunch of 'default' const ints that all = 3, and why use multiple when you can use one?  Just a small memory optimization.  I'll revert it.